src/Microsoft.FeatureManagement/FeatureFilters/Recurrence/RecurrenceEvaluator.cs

```diff
@@ -0,0 +1,1441 @@
+                {
+                    return false;
+                }
+
+                if (!TryValidateRecurrenceRange(settings, out paramName, out reason))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static bool TryValidateGeneralRequiredParameter(TimeWindowFilterSettings settings, out string paramName, out string reason)
+        {
+            Recurrence recurrence = settings.Recurrence;
+
+            paramName = null;
+
+            reason = null;
+
+            if (settings.Start == null)
+            {
+                paramName = nameof(settings.Start);
+
+                reason = RequiredParameter;
+
+                return false;
+            }
+
+            if (settings.End == null)
+            {
+                paramName = nameof(settings.End);
+
+                reason = RequiredParameter;
+
+                return false;
+            }
+
+            if (recurrence.Pattern == null)
+            {
+                paramName = $"{nameof(settings.Recurrence)}.{nameof(recurrence.Pattern)}";
+
+                reason = RequiredParameter;
+
+                return false;
+            }
+
+            if (recurrence.Range == null)
+            {
+                paramName = $"{nameof(settings.Recurrence)}.{nameof(recurrence.Range)}";
+
+                reason = RequiredParameter;
+
+                return false;
+            }
+
+            if (recurrence.Pattern.Type == null)
+            {
+                paramName = $"{nameof(settings.Recurrence)}.{nameof(recurrence.Pattern)}.{nameof(recurrence.Pattern.Type)}";
+
+                reason = RequiredParameter;
+
+                return false;
+            }
+
+            if (recurrence.Range.Type == null)
+            {
+                paramName = $"{nameof(settings.Recurrence)}.{nameof(recurrence.Range)}.{nameof(recurrence.Range.Type)}";
+
+                reason = RequiredParameter;
+
+                return false;
+            }
+
+            if (settings.End.Value - settings.Start.Value <= TimeSpan.Zero)
+            {
+                paramName = nameof(settings.End);
+
+                reason = OutOfRange;
+
+                return false;
+            }
+
+            return true;
+        }
+
+        private static bool TryValidateRecurrencePattern(TimeWindowFilterSettings settings, out string paramName, out string reason)
+        {
+            paramName = null;
+
+            reason = null;
+
+            if (!TryValidateInterval(settings, out paramName, out reason))
+            {
+                return false;
+            }
+
+            string patternType = settings.Recurrence.Pattern.Type;
+
+            if (string.Equals(patternType, Daily, StringComparison.OrdinalIgnoreCase))
+            {
+                if (!TryValidateDailyRecurrencePattern(settings, out paramName, out reason))
```

Comment:
```
if (!TryValidateDailyRecurrencePattern(settings, out paramName, out reason))
{
    return false;
}
```
The code could be simplified as the following:
```
return TryValidateDailyRecurrencePattern(settings, out paramName, out reason);
```

Reply:
Resolved.